pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1991,7 +1989,10 @@ func convertGateways(r configContext) ([]config.Config, map[parentKey][]*parentI
 				reportListenerAttachedRoutes(i, obj, pri.AttachedRoutes)
 			}
 			gwMap[ref] = append(gwMap[ref], pri)
-			result = append(result, gatewayConfig)
+
+			if programmed {
```

Comment:
Hmm. A bit worried by this actually. Do we really want to send no config?

There is a huge difference in excluding a listener vs including it with invalid TLS config (in terms of Envoy)

Consider:

```
listeners:
- host: '*.example.com'
  tls: valid-secret
- host: 'foo.example.com'
  tls: invalid-secret
```

In one case, all traffic to foo.example.com is blocked. In the other, it falls into `*.example.com`.

Even worse, if foo.example.com transitioned from valid -> invalid, in the old way (second config) I believe it would still work (caching last good config).

Reply:
Well this is how it works today - we send no config in the invalid case, because of the continue statement, above:
```
			if !ok {
				continue
			}
```
This PR isn't changing that, it's just allowing the code to go further so that it properly updates the status of the associated k8s resources.

I could look into changing it to always send comfig, even invalid ones, but probably better in a followup PR?

Reply:
ack, probably good as is then. will take another look